### PR TITLE
Fix: some competition calendar things

### LINKED ISSUE #41: Calendar doesnt work correctly when changing years
I presume because of the switch to server side calls only, the calendar doesnt show the correct competitions when changing the year filter.

### PR DESCRIPTION
**Pull request by dorianim**
- [X] use league_group for calendar (needs https://github.com/speedclimbing/microservices/pull/35)
- [X] fix calendar behavior

Fixes #41

### CODE REVIEW

File: src/routes/competition/+page.svelte [hunk 1]
```diff
@@ -1,41 +1,88 @@
...
+	<DebouncedInput
+		type="text"
+		placeholder="Competition Name"
+		inputClass="rounded-sm font-Raleway bg-black/5"
+		bind:value={name}
+	/>
+	<SelectFilter
+		bind:value={year}
+		options={data.seasons}
+		disabled={viewCalendar}
```
Comment (lamorak13) [inline src/routes/competition/+page.svelte:51]: I wouldnt disable the year select if the calendar is active. If the other selects are active this one should be as well

File: src/routes/competition/+page.svelte [hunk 2]
```diff
@@ -1,41 +1,88 @@
...
+	}
+
+	$: {
+		const currentDate = new Date();
+		if (!viewCalendar && year === currentDate.getFullYear()) {
+			calendarDate = currentDate;
+		} else if (!viewCalendar && year !== currentDate.getFullYear()) {
+			calendarDate = undefined;
+		} else if (viewCalendar && calendarDate && year !== calendarDate.getFullYear()) {
+			year = calendarDate.getFullYear();
```
Comment (lamorak13) [inline src/routes/competition/+page.svelte:36]: Maybe there is some way to refactor this so that it is easier to understand what happens

**Review by lamorak13 — CHANGES_REQUESTED** [review]
Looks very good so far

File: src/routes/competition/+page.svelte (hunk 2 above)
Comment (dorianim) [reply src/routes/competition/+page.svelte]: I rewrote this part, hope it is better to  understand now :)

**Review by lamorak13 — APPROVED** [review]

**Pull request merged**